[UnifiedPDF][iOS] Show find indicator for the current match

pxlcoder opened the pull request:
#### 9a109e4656341e3e8e19621125dd918afa1a139b
<pre>
[UnifiedPDF][iOS] Show find indicator for the current match
<a href="https://bugs.webkit.org/show_bug.cgi?id=280574">https://bugs.webkit.org/show_bug.cgi?id=280574</a>
<a href="https://rdar.apple.com/136903474">rdar://136903474</a>

Reviewed by Wenson Hsieh.

Create and display a text indicator for the current match. Additionally, scroll
to reveal the current match as it is changed.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
(WebKit::PDFPluginBase::textIndicatorForTextMatch):
(WebKit::PDFPluginBase::scrollToRevealTextMatch):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::scrollToRevealTextMatch):

On iOS, the native scroll view must be scrolled to reveal the current match.

(WebKit::UnifiedPDFPlugin::textIndicatorForTextMatch):
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::textIndicatorForTextMatch):
(WebKit::PluginView::scrollToRevealTextMatch):
* Source/WebKit/WebProcess/Plugins/PluginView.h:
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
(WebKit::WebFoundTextRangeController::decorateTextRangeWithStyle):
(WebKit::WebFoundTextRangeController::scrollTextRangeToVisible):
(WebKit::WebFoundTextRangeController::redraw):
(WebKit::WebFoundTextRangeController::createTextIndicatorForPDFRange):
(WebKit::WebFoundTextRangeController::setTextIndicatorWithPDFRange):
(WebKit::WebFoundTextRangeController::flashTextIndicatorAndUpdateSelectionWithPDFRange):
(WebKit::WebFoundTextRangeController::rectsForTextMatchesInRect):

Do not paint a hole for the current match.

(WebKit::WebFoundTextRangeController::frameForFoundTextRange const):

Factor out common logic to determine the local frame for a match.

(WebKit::WebFoundTextRangeController::documentForFoundTextRange const):
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.h:

Canonical link: <a href="https://commits.webkit.org/284437@main">https://commits.webkit.org/284437@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/28377b2f098eeafb7661462b047f38c6a1c2b0f0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69333 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48733 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22006 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73415 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20491 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71450 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56534 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20341 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55144 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13606 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72399 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44463 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59861 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35622 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41129 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17291 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18868 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63071 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17636 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75127 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13315 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16859 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62811 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13354 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59944 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62717 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10732 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4345 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10593 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44537 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45611 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46806 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45352 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
```diff
@@ -415,6 +415,8 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
     Vector<WebFoundTextRange::PDFData> findTextMatches(const String&, WebCore::FindOptions) final;
     Vector<WebCore::FloatRect> rectsForTextMatch(WebFoundTextRange::PDFData) final;
+    RefPtr<WebCore::TextIndicator> textIndicatorForTextMatch(WebFoundTextRange::PDFData, WebCore::TextIndicatorPresentationTransition) final;
+    void scrollToRevealTextMatch(WebFoundTextRange::PDFData) final;
```
Is there a reason we use pass-by-value instead of const ref or rvalue reference here? (since the struct's size is larger than 8 bytes)

pxlcoder replied:
No reason – fixed to use a const ref!

pxlcoder committed:
[UnifiedPDF][iOS] Show find indicator for the current match
https://bugs.webkit.org/show_bug.cgi?id=280574
rdar://136903474

Reviewed by Wenson Hsieh.

Create and display a text indicator for the current match. Additionally, scroll
to reveal the current match as it is changed.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
(WebKit::PDFPluginBase::textIndicatorForTextMatch):
(WebKit::PDFPluginBase::scrollToRevealTextMatch):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::scrollToRevealTextMatch):

On iOS, the native scroll view must be scrolled to reveal the current match.

(WebKit::UnifiedPDFPlugin::textIndicatorForTextMatch):
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::textIndicatorForTextMatch):
(WebKit::PluginView::scrollToRevealTextMatch):
* Source/WebKit/WebProcess/Plugins/PluginView.h:
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
(WebKit::WebFoundTextRangeController::decorateTextRangeWithStyle):
(WebKit::WebFoundTextRangeController::scrollTextRangeToVisible):
(WebKit::WebFoundTextRangeController::redraw):
(WebKit::WebFoundTextRangeController::createTextIndicatorForPDFRange):
(WebKit::WebFoundTextRangeController::setTextIndicatorWithPDFRange):
(WebKit::WebFoundTextRangeController::flashTextIndicatorAndUpdateSelectionWithPDFRange):
(WebKit::WebFoundTextRangeController::rectsForTextMatchesInRect):

Do not paint a hole for the current match.

(WebKit::WebFoundTextRangeController::frameForFoundTextRange const):

Factor out common logic to determine the local frame for a match.

(WebKit::WebFoundTextRangeController::documentForFoundTextRange const):
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.h:

Canonical link: https://commits.webkit.org/284437@main
